core/src/translate/query.rs

```diff
@@ -173,26 +173,36 @@ fn translate_table_factor(sql_table_factor: &SqlTableFactor) -> Result<TableFact
     match sql_table_factor {
         SqlTableFactor::Table {
             name, alias, args, ..
-        } if translate_object_name(name)?.to_uppercase() == "SERIES" && args.is_some() => {
-            let name = match alias {
-                Some(SqlTableAlias { name, .. }) => name.value.to_owned(),
-                None => translate_object_name(name)?,
+        } => {
+            let object_name = translate_object_name(name)?.to_uppercase();
+            let alias = translate_table_alias(alias);
+            let alias_or_name = || -> TableAlias {
+                match &alias {
+                    Some(alias) => alias.to_owned(),
+                    None => TableAlias {
+                        name: object_name.to_owned(),
+                        columns: Vec::new(),
+                    },
+                }
             };
 
-            Ok(TableFactor::Series {
-                alias: TableAlias {
-                    name,
-                    columns: Vec::new(),
-                },
-                size: translate_table_args(args)?,
-            })
-        }
-        SqlTableFactor::Table { name, alias, .. } => {
-            Ok(TableFactor::Table {
-                name: translate_object_name(name)?,
-                alias: translate_table_alias(alias),
-                index: None, // query execution plan
-            })
+            match object_name.as_str() {
+                "SERIES" if args.is_some() => Ok(TableFactor::Series {
+                    alias: alias_or_name(),
+                    size: translate_table_args(args)?,
+                }),
+                "GLUE_TABLES" | "GLUE_TABLE_COLUMNS" => Ok(TableFactor::Dictionary {
+                    dict: translate_dictionary_name(object_name.as_str())?,
+                    alias: alias_or_name(),
+                }),
```

Comment:
Here it looks that we can assign`Dictionary::GlueTables` and `Dictionary::GlueTableColumns` directly without adding `translate_dictionary_name` function. Then, we don't need to add `UnreachableDictionary` error.